<launch>
    <!-- SYSTEM MODEL -->
    <!-- vehicle parameters -->
    <param name="m" type="double" value="1.98" />
    <param name="I_z" type="double" value="0.24" />
    <param name="L_a" type="double" value="0.125" />
    <param name="L_b" type="double" value="0.125" />
    <!-- control parameters -->
    <param name="input_gain" type="double" value="0.3" />
    <param name="air_drag_coeff" type="double" value="0.1308" />
    <param name="friction" type="double" value="0.1711" />
    <!-- tire parameters -->
    <param name="tire_model/B" type="double" value="0.3" />
    <param name="tire_model/C" type="double" value="1.25" />
    <param name="tire_model/mu" type="double" value="0.234" />

    <!-- IMU -->
    <node pkg="myahrs_driver" type="myahrs_driver" name="myahrs_driver">
        <param name="port" value="/dev/ttyACM0" />
        <param name="baud_rate" value="115200" />
    </node>

    <!-- ARDUINO -->
    <!--  * encoders and ultrasound sensors -->
    <node pkg="rosserial_python" type="serial_node.py" name="arduino_node" >
        <param name="port" value="/dev/ttyUSB0" />
    </node>

    <!-- OBSERVER-->
    <node pkg="barc" type="state_estimation_DynBkMdl.py" name="state_estimation">
        <!-- v_x estimation sample time-->
        <param name="dt_v_enc" type="double" value="0.2" />

        <!-- ekf properties -->
        <param name="q_std" type="double" value="0.1" />
        <param name="r_std" type="double" value="0.1" />
        <param name="v_x_min" type="double" value="0.3" />
    </node>

    <!-- LOW LEVEL -->
    <node pkg="barc" type="controller_low_level.py" name="low_level_controller" />
</launch>
